14.0 Add module:  sale quotation template product multi add

ilyasProgrammer opened the pull request:
Similar to **sale_product_multi_add**, but for Quotation Template (sale.order.template) model.

ilyasProgrammer committed:
[14.0][ADD] new module sale_quotation_template_product_multi_add

rousseldenis commented on sale_quotation_template_product_multi_add/wizards/qt_add_products.py:
```diff
@@ -0,0 +1,72 @@
+
+
+class QTAddProducts(models.TransientModel):
+    _name = "qt.add.products"
```
Better to use an explicit name

rousseldenis commented on sale_quotation_template_product_multi_add/wizards/qt_add_products.py:
```diff
@@ -0,0 +1,72 @@
+    _name = "qt.add.products"
+    _description = "Quotation Template Add Products"
+
+    products = fields.Many2many(comodel_name="product.product")
```
The convention would require 'product_ids' instead (same for here under)

rousseldenis commented on sale_quotation_template_product_multi_add/wizards/qt_add_products_view.xml:
```diff
@@ -0,0 +1,55 @@
+        <field name="model">qt.add.products</field>
+        <field name="arch" type="xml">
+            <form string="Products">
+                <field name="products" />
```
Sometimes it's better to add a short help text for user. What do you think ?

rousseldenis commented on sale_quotation_template_product_multi_add/wizards/qt_add_products.py:
```diff
@@ -0,0 +1,72 @@
+    def create_items(self):
+        for wizard in self:
+            for product in wizard.products:
+                self.env["qt.add.products.items"].create(
```
It's better to use create multi functionality

rousseldenis reviewed (changes requested):
Code review. Some comments

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 80.0% (12 of 15 strings)

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_start_end_dates
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_start_end_dates/it/

rousseldenis commented on sale_amount_payment_link/tests/test_sale_amount_payment_link.py:
```diff
@@ -1,11 +1,13 @@
 from odoo.addons.account.tests.common import AccountTestInvoicingCommon
 
 
+@tagged("post_install")
```
@ilyasProgrammer Could you remove unrelated module modifications ?

rousseldenis replied:
Ok, I see, it's to fix tests. But we can do it in another PR in order to merge it fast. Could you do it ?

ilyasProgrammer replied:
ok

ilyasProgrammer replied:
https://github.com/OCA/sale-workflow/pull/1989

francesco-ooops replied:
@ilyasProgrammer rebase?

ilyasProgrammer committed:
[14.0][TEST] rollback tests modification

ilyasProgrammer committed:
[14.0][TEST] sale_amount_payment_link sale_invoice_plan  Fix tests

oca-travis committed:
[UPD] Update sale_tier_validation.pot

OCA-git-bot committed:
sale_amount_payment_link 14.0.1.0.1

OCA-git-bot committed:
sale_invoice_plan 14.0.1.0.2

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: sale-workflow-14.0/sale-workflow-14.0-sale_tier_validation
Translate-URL: https://translation.odoo-community.org/projects/sale-workflow-14-0/sale-workflow-14-0-sale_tier_validation/

ilyasProgrammer commented:
this PR was broken. Moved here https://github.com/OCA/sale-workflow/pull/1990
